pkg/util/ready/ready.go

```diff
@@ -24,6 +29,30 @@ import (
 	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
 )
 
+// IsNetworkUnreachableError return true if this is a connection error
+func IsNetworkUnreachableError(err error) bool {
```

Comment:
The naming of this function is a little confusing to me ...especially with that check for the `x509.UnknownAuthorityError`. It might be worth having a look at how `go-autorest` does this in `isTemporaryNetworkError(err)`.

```
// returns true if the specified error is a temporary network error or false if it's not.
// if the error doesn't implement the net.Error interface the return value is true.
func isTemporaryNetworkError(err error) bool {
	if netErr, ok := err.(net.Error); !ok || (ok && netErr.Temporary()) {
		return true
	}
	return false
}
```

Reply:
yip, i looked there, but want to  catch some more errors... maybe name it isRetryableError()?

Reply:
maybe `isRetryableConnectionError`?

Reply:
I don't want to refactor the different error handling points into one, for several reasons:
* golang error nesting differs a lot and really hampers this approach over time
* we lose understanding of what errors are actually happening at each point and why